src/components/views/rooms/RoomTile2.tsx

```diff
@@ -86,10 +88,19 @@ export default class RoomTile2 extends React.Component<IProps, IState> {
             hover: false,
             notificationState: this.getNotificationState(),
         };
+
+        // TODO: We shouldn't have to listen to every room update
+        // We don't have a model which works in a better way though.
+        MatrixClientPeg.get().on("Room.receipt", this.tryUpdateIfRoomMatches);
+        MatrixClientPeg.get().on("Room.timeline", this.tryUpdateIfRoomMatches);
+        MatrixClientPeg.get().on("Event.decrypted", this.tryUpdateIfRoomMatches);
+        MatrixClientPeg.get().on("Room.redaction", this.tryUpdateIfRoomMatches);
```

Comment:
Any reason the 3 `Room.` event handlers are not bound on the room eventemitter itself to fire less?

Reply:
Honestly just didn't expect it to work. If it does, then great! The TODO comment can go away with it.